kasa/discover.py

```diff
@@ -334,9 +362,8 @@ async def discover_single(
         if ip in protocol.discovered_devices:
             dev = protocol.discovered_devices[ip]
             dev.host = host
-            # Call device update on devices that have children
-            if update_parent_devices and dev.has_children:
-                await dev.update()
+            if httpx_asyncclient and hasattr(dev.protocol._transport, "http_client"):
+                dev.protocol._transport.http_client = httpx_asyncclient  # type: ignore[union-attr]
```

Comment:
This feels fragile construct. As we are not anymore doing a forced update on discovered devices, maybe the downstream users should just adjust the connection parameters of the device instance to their liking after the discovery?

Reply:
Ok that makes sense but to avoid HA creating unnecessary httpx_clients for the legacy devices we need a way to indicate to the consumer whether to set an http_client or not.  I think the easiest way would be to add a property `uses_http` to the connection parameters which the consumer can then check before setting the `http_client`.  Agree?

Reply:
Fixed with uses_http added to `DeviceConfig`